### PR TITLE
Gather Prometheus metrics from tasks

### PR DESCRIPTION
**Pull request by philipnrmn**
## High-level description

This PR enables Prometheus metrics collection from ports on tasks which specify `DCOS_METRICS_FORMAT=prometheus` in their labels.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-3717](https://jira.mesosphere.com/browse/DCOS_OSS-3717) Add Mesos service discovery to the Telegraf prometheus metrics input plugin

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/telegraf/compare/a88d78350c9244336607e37cc808bb76a40e8b89...ba18887f5d6358e3eb73d5df7e16a2c19c3cbfda)
  - [x] Test Results: [link to CI job test results for component](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-cluster-ops/job/telegraf/job/telegraf-dcos-pulls/161/)
  - [x] Code Coverage (if available): [link to code coverage report](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-cluster-ops/job/telegraf/job/telegraf-dcos-pulls/161/)

### CODE REVIEW

**Comment by philipnrmn** [comment]
@mesosphere-mergebot bump-ee

**Comment by philipnrmn** [comment]
@mesosphere-mergebot bump-ee

**Comment by philipnrmn** [comment]
@mesosphere-mergebot bump-ee

**Comment by philipnrmn** [comment]
@mesosphere-mergebot bump-ee

**Comment by philipnrmn** [comment]
@mesosphere-mergebot bump-ee

**Comment by philipnrmn** [comment]
@mesosphere-mergebot label ready for review

**Comment by philipnrmn** [comment]
@branden and @gracedo, would you review this when you have a moment please?

File: gen/dcos-config.yaml [hunk 1]
```diff
@@ -1571,6 +1571,12 @@ package:
         percentile_limit = 1000
       # Read metrics from Admin Router Nginx Prometheus endpoint.
       [[inputs.prometheus]]
+        ## The URL of the local mesos agent
+        mesos_agent_url = "http://$DCOS_NODE_PRIVATE_IP:5051"
+        ## The period after which requests to mesos agent should time out
+        mesos_timeout = "10s"
+        ## The user agent to send with requests
+        user_agent = "telegraf-prometheus"
```
Comment (gracedo) [inline gen/dcos-config.yaml]: `Telegraf-prometheus` (but I don't block on it, can fix in another dcos pr!)

File: gen/dcos-config.yaml [hunk 2]
```diff
@@ -1683,6 +1689,13 @@ package:
...
```
Comment (gracedo) [inline gen/dcos-config.yaml]: `Telegraf-prometheus`

**Comment by orsenthil** [comment]
@mesosphere-mergebot label work in progress

**Comment by orsenthil** [comment]
Setting this to work in progress as that seems like the right state.

File: CHANGES.md [hunk 3]
```diff
@@ -31,6 +31,7 @@ This change also aligned the authentication architectures between DC/OS Enterpri
 
 * Prometheus-format metrics can be gathered from tasks (DCOS_OSS-3717)
 
+
```
Comment (philipnrmn) [inline CHANGES.md]: I added this to trigger the tests (but it is a good change nonetheless)

**Comment by philipnrmn** [comment]
rebased to address dcos-e2e issue

**Pull request merged**